be/src/exec/pipeline/crossjoin/cross_join_left_operator.cpp

```diff
@@ -325,17 +326,26 @@ void CrossJoinLeftOperatorFactory::_init_row_desc() {
     }
 }
 
+OperatorPtr CrossJoinLeftOperatorFactory::create(int32_t degree_of_parallelism, int32_t driver_sequence) {
+    return std::make_shared<NLJoinProbeOperator>(this, _id, _plan_node_id, driver_sequence, _join_op,
```

Comment:
In  CrossJoinLeftOperatorFactory, a NLJoinProbeOperator is created, is the CrossJoinLeftOperator useless? if so, remove the deprecated files. reserve the operator and operator factory the same root name either, not both.

Reply:
Two steps to replace the `CrossJoinOperator` with `NestloopOperator`:
1. Create `NestloopOperator` in `CrossJoinOperatorFactory`
2. Rename the `CrossJoinOperatorFactory to `NestloopJoinOperatorFactory`
3. Remove all cross-join related code

Reply:
CrossJoinOperator is useless, not instantiated, not compatibility issue, so it can be removed safely.